middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -619,6 +619,22 @@ let close_primitive acc env ~let_bound_var named (prim : Lambda.primitive) ~args
     in
     (* Since raising of an exception doesn't terminate, we don't call [k]. *)
     Expr_with_acc.create_apply_cont acc apply_cont
+  | (Pmakeblock _ | Pmakefloatblock _ | Pmakearray _), [] ->
+    (* Special case for liftable empty blocks *)
+    begin
+      match prim with
+      | Pmakeblock (tag, _, _) when tag <> 0 ->
```

Comment:
Could you also add a fatal error on the `Pmakefloatblock` case, which is more or less a `Pmakeblock` with a different tag ?

Reply:
There is no tag info on `Pmakefloatblock` because it acts as a special case of `Pmakeblock(Obj.floatarray_tag, _, _)`. As with empty arrays, it's correct to compile it to a block with tag 0, but as with regular blocks with non-zero tags this case is not supposed to happen at all currently.